employees/tests/test_views.py

```diff
@@ -217,3 +212,151 @@ def test_delete_report_view_should_delete_report_on_post(self):
+                "task_activities": self.report.task_activities.pk,
+            },
+        )
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Report.objects.all().count(), 2)
+
+    def test_custom_report_list_view_should_not_add_new_report_on_post_if_form_is_invalid(self):
+        response = self.client.post(
+            path=self.url, data={"description": "Some description", "project": self.project.pk, "work_hours": "8:00"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(Report.objects.all().count(), 1)
+        self.assertIsNotNone(response.context_data["form"].errors)
+
+    def test_get_queryset_method_should_return_queryset_containing_all_of_current_users_reports(self):
+        other_user = UserFactory()
+        other_project = ProjectFactory()
+        other_project.members.add(other_user)
+        other_user_report = ReportFactory(author=other_user, project=other_project)
+
+        ReportFactory(author=self.user, project=self.project)
+        ReportFactory(author=self.user, project=self.project)
+        response = self.client.get(path=self.url)
+        queryset = response.context_data["object_list"]
+        self.assertEqual(len(queryset), 3)
+        self.assertFalse(other_user_report in queryset)
+        for report in self.user.projects.all():
+            self.assertContains(response, report)
+
+    def test_custom_report_list_add_project_method_should_register_current_user_as_project_member(self):
+        self.new_project.members.add(self.user)
+        self.assertTrue(self.user in self.new_project.members.all())
+
+    def test_custom_report_list_view_should_add_user_to_project_selected_in_project_join_form_on_join(self):
+        response = self.client.post(path=self.url, data={"projects": self.new_project.pk, "join": "join"})
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.user in self.new_project.members.all())
+
+    def test_custom_report_list_view_should_not_add_user_to_project_if_join_not_in_request(self):
+        response = self.client.post(
+            path=self.url,
+            data={
+                "date": datetime.datetime.now().date(),
+                "description": "Some description",
+                "project": self.project.pk,
+                "work_hours": "8:00",
+                "task_activities": self.report.task_activities.pk,
+                "projects": self.new_project.pk,
+            },
+        )
+        self.assertTrue(response.status_code, 302)
+        self.assertIn(self.new_project, Project.objects.all())
+        self.assertTrue(self.user not in self.new_project.members.all())
+
+    def test_custom_report_list_view_should_handle_no_project_being_selected_in_project_form_on_post(self):
+        response = self.client.post(path=self.url, data={"join": "join"})
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(response.context_data["form"].errors)
```

Comment:
You could check what error is exactly returned here, and if it is the one that you expect.